flood: repackage using buildNpmPackage; flood: 4.7.0 -> unstable-2023-06-03

thiagokokada opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Related issue: #229475

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

thiagokokada committed:
flood: repackage using buildNpmPackage

Related issue: #229475

Co-authored-by: Winter <winter@winter.cafe>

thiagokokada commented:
Pinging @winterqt here, I got the `default.nix` file directly from the [documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md#buildnpmpackage-javascript-buildnpmpackage) and I imagine the only reason this was not done before is because @winterqt didn't realize we already had this packaged in Nixpkgs.

SuperSandro2000 commented on pkgs/applications/networking/p2p/flood/default.nix:
```diff
@@ -0,0 +1,27 @@
+  NODE_OPTIONS = "--openssl-legacy-provider";
+
+  meta = with lib; {
+    description = "A modern web UI for various torrent clients with a Node.js backend and React frontend";
```
```suggestion
    description = "Modern web UI for various torrent clients with a Node.js backend and React frontend";
```

thiagokokada replied:
Done.

SuperSandro2000 commented on pkgs/applications/networking/p2p/flood/default.nix:
```diff
@@ -0,0 +1,27 @@
+
+  npmDepsHash = "sha256-tuEfyePwlOy2/mOPdXbqJskO6IowvAP4DWg8xSZwbJw=";
+
+  # The prepack script runs the build script, which we'd rather do in the build phase.
+  npmPackFlags = [ "--ignore-scripts" ];
```
Isn't this the default? Can you check that?

thiagokokada replied:
Done.

winterqt reviewed (changes requested):
> Pinging @winterqt here, I got the `default.nix` file directly from the [documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md?rgh-link-date=2023-07-25T20%3A14%3A35Z#buildnpmpackage-javascript-buildnpmpackage) and I imagine the only reason this was not done before is because @winterqt didn't realize we already had this packaged in Nixpkgs.

I just keep forgetting to actually upstream this (I use a version of this in my own configuration).

This was the first package I successfully packaged with this tooling once I wrote it, and has been the de-facto package for testing v2 lockfiles, heh.

~~Given this, I'd appreciate a `Co-authored-by: Winter <winter@winter.cafe>` if you have to push this again to fix the comments above, but I'm not too worried about it.~~

Thank you for the PR!

thiagokokada commented on pkgs/applications/networking/p2p/flood/default.nix:
```diff
@@ -0,0 +1,27 @@
+
+buildNpmPackage rec {
+  pname = "flood";
+  version = "4.7.0";
```
I tried to backport this commit but it fails to apply cleanly in the current version. We can probably backport more patches to make this work, but it seems that it is better to just override this locally if you need to work with modern qBittorrent versions for now.

thiagokokada replied:
Another option would be to just package unstable for now, WDYT @winterqt ?

thiagokokada replied:
BTW, this release is from 2021, so ancient times in JS community. Probably have some security issues already.

So maybe should we go with unstable?

thiagokokada replied:
Well, I tried to build the latest [commit](https://github.com/jesec/flood/commit/2b652f8148dab7134eeeb201b9d81dd6b8bda074), and while it builds successfully I just got a blank screen and some errors in the developer console.

So I don't know, at least version 4.7.0 seems to works fine.

thiagokokada replied:
Pushed the bump in a separate commit, so if you want to bump the version to the current development version just say so and we can commit it instead of 4.7.0.

thiagokokada commented:
> Given this, I'd appreciate a Co-authored-by: Winter <winter@winter.cafe> if you have to push this again to fix the comments above, but I'm not too worried about it.

Of course, done :smile:.

thiagokokada committed:
flood: 4.7.0 -> unstable-2023-06-03

Bumping to current development version to fix some deprecation warns in
Node and also issues with newer versions of qBittorrent.

thiagokokada commented:
Pinging @winterqt.

winterqt commented:
Sorry, I'll have the time to look at this in a day or so -- I haven't forgotten!

winterqt reviewed (approved):
Please fix the merge conflicts.
